fe/fe-core/src/main/java/com/starrocks/lake/snapshot/ClusterSnapshotConfig.java

```diff
@@ -0,0 +1,266 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+package com.starrocks.lake.snapshot;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.fasterxml.jackson.annotation.JsonValue;
+import com.fasterxml.jackson.core.JsonParser;
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.core.type.TypeReference;
+import com.fasterxml.jackson.databind.DeserializationContext;
+import com.fasterxml.jackson.databind.JsonDeserializer;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
+import com.fasterxml.jackson.dataformat.yaml.YAMLFactory;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+import java.io.File;
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+public class ClusterSnapshotConfig {
+    private static final Logger LOG = LogManager.getLogger(ClusterSnapshotConfig.class);
+
+    public static class Frontend {
+        public static enum FrontendType {
+            FOLLOWER,
+            OBSERVER;
+
+            @JsonCreator
```

Comment:
Can we use json? This way, we can unify the development to GSON and reduce the types of configuration for users.

Reply:
json do not support comment and is suitable for machine parsing but not for human reading, it is not suitable for configuration file.
yaml support comment and suitable for both machine parsing and human reading, it is a better new generation configuration file.
by the way, yaml can be handled by Jackson, which is used in many places in FE.